### PR TITLE
#19 - Added PackageMetadata to service index

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #19 
Added PackageMetadata to service index & NuGetServiceIndexTest refactoring

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 PLease assign me, I'm available

File: src/main/java/com/artpie/nuget/http/NuGet.java [hunk 1]
```diff
@@ -105,18 +105,20 @@ public Response response(
...
     private Resource resource(final String path) {
         final PackagePublish publish = new PackagePublish(this.storage);
+        final PackageMetadata metadata = new PackageMetadata();
         final PackageContent content = new PackageContent(this.storage);
         return new RoutingResource(
             path,
             new ServiceIndex(
                 Arrays.asList(
                     new RouteService(this.url, publish, "PackagePublish/2.0.0"),
+                    new RouteService(this.url, metadata, "RegistrationsBaseUrl/Versioned"),
```
Comment (paulodamaso) [inline src/main/java/com/artpie/nuget/http/NuGet.java]: @olegmoz Shouldn't we be using one of the types described in https://docs.microsoft.com/EN-US/nuget/api/registration-base-url-resource#versioning ?

File: src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java [hunk 2]
```diff
@@ -83,16 +86,26 @@ void shouldGetIndex() {
...
+                            new AllOf<>(
                                 Arrays.asList(
-                                    new IsService(
-                                        "PackagePublish/2.0.0",
-                                        String.format("%s/package", this.url)
-                                    ),
-                                    new IsService(
-                                        "PackageBaseAddress/3.0.0",
-                                        String.format("%s/content", this.url)
+                                    new JsonHas("version", new JsonValueIs("3.0.0")),
```
Comment (paulodamaso) [inline src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java]: @g4s8 Wouldn't be a good idea to move some of these matchers up so we can use them into other artipie projects?

File: src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java [hunk 3]
```diff
@@ -83,16 +86,26 @@ void shouldGetIndex() {
...
+                                    new JsonHas(
+                                        "resources",
+                                        new JsonContains(
+                                            new IsService(
+                                                "PackagePublish/2.0.0",
+                                                String.format("%s/package", this.url)
+                                            ),
+                                            new IsService(
+                                                "RegistrationsBaseUrl/Versioned",
```
Comment (paulodamaso) [inline src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java]: @olegmoz PLease see above comment

File: src/main/java/com/artpie/nuget/http/NuGet.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artpie/nuget/http/NuGet.java]: @paulodamaso Actually it is undocumented, but you may find it in NuGet source code https://github.com/NuGet/NuGet.Client that the name I use is common fallback if NuGet client does not find expected version. I have tested is with latest client version 5.4 and it works. Probably we will need to support proper versions later, but for minimum usage it's enough.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) not enough funds available in the project, can't set budget of job [#62](https://github.com/artipie/nuget-adapter/issues/62), see [§21](http://www.zerocracy.com/policy.html#21); @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) will get no money on completion; in order to fix that, add funds to the project and assign the job again

File: src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java (hunk 2 above)
Comment (olegmoz) [reply src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java]: @paulodamaso Json matchers I am using here are actually a library, so anyone can use them. But maybe matcher for JSON response will be useful in other places. Will extract it, thanks!

File: src/main/java/com/artpie/nuget/http/NuGet.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artpie/nuget/http/NuGet.java]: Added issue: https://github.com/artipie/http/issues/93

**Comment by olegmoz** [comment]
@paulodamaso ping

**Comment by g4s8** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/nuget-adapter/issues/62#issuecomment-604304592))

@g4s8 The user @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) resigned from [#62](https://github.com/artipie/nuget-adapter/issues/62), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
This pull request [#62](https://github.com/artipie/nuget-adapter/issues/62) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/566c0dd9-3c00-47bf-9d4d-de391526feda); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@olegmoz Seems good to me, okay to merge

**Comment by paulodamaso** [comment]
@g4s8 please merge

**Comment by 0crat** [comment]
Job was finished in 4 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Comment by paulodamaso** [comment]
@g4s8 Didn'r received payment for this one, please take a look

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/nuget-adapter/issues/62#issuecomment-612862896))

@g4s8 This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#62](https://github.com/artipie/nuget-adapter/issues/62) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fnuget-adapter%2362%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@0crat pay 15min @paulodamaso

**Pull request merged**